modules/publicize/publicize.php

```diff
@@ -107,33 +107,210 @@ function __construct() {
 		// then check meta and publicize based on that. stage 3 implemented on wpcom
 		add_action( 'transition_post_status', array( $this, 'flag_post_for_publicize' ), 10, 3 );
 		add_action( 'save_post', array( &$this, 'save_meta' ), 20, 2 );
+
+		// Default checkbox state for each Connection
+		add_filter( 'publicize_checkbox_default', array( $this, 'publicize_checkbox_default' ), 10, 4 );
+
+		// Alter the "Post Publish" admin notice to mention the Connections we Publicized to.
 		add_filter( 'post_updated_messages', array( $this, 'update_published_message' ), 20, 1 );
 
 		// Connection test callback
 		add_action( 'wp_ajax_test_publicize_conns', array( $this, 'test_publicize_conns' ) );
 	}
 
+/*
+ * Services: Facebook, Twitter, etc.
+ */
+
 	/**
-	* Functions to be implemented by the extended class (publicize-wpcom or publicize-jetpack)
-	*/
-	abstract function get_connection_id( $connection );
+	 * Get services for the given blog and user.
+	 *
+	 * Can return all avaialable services or just the ones with an active connection.
```

Comment:
Typo, `avaialable` (missed it at first, only noticed now when looking at the WP.com side diff)